modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/raft/MetaStorageWriteHandler.java

```diff
@@ -447,4 +442,27 @@ public void result(@Nullable Serializable res) {
             closure.result(res);
         }
     }
+
+    private List<byte[]> collectEvictionCandidateKeys(HybridTimestamp evictionTimestamp) {
```

Comment:
Do we have a test for this new logic?

Reply:
Add **org.apache.ignite.internal.metastorage.impl.ItIdempotentCommandCacheTest#testEvictIdempotentCommandsCacheAfterCompaction**